core/src/main/scripts/mutationalSignatures.py

```diff
@@ -0,0 +1,562 @@
+    for idx, row in maf.iterrows():  
+        if row["Tumor_Seq_Allele1"] != row["Reference_Allele"] \
+            and regexp.fullmatch(row["Tumor_Seq_Allele1"]):
+            return "Tumor_Seq_Allele1"
+        elif row["Tumor_Seq_Allele2"] != row["Reference_Allele"] \
+            and regexp.fullmatch(row["Tumor_Seq_Allele2"]):
+            return "Tumor_Seq_Allele2"
+    raise ValueError("Alternative Allele in maf file could not be determined. "
+                     "Please use the --alt-allele parameter.")
+
+
+def preprocess_maf(mut_file: str, tmp_dir: str, args: argparse.Namespace) -> tuple:
+    """Preprocess the cBioPortal mutations data file for matrix generation 
+    
+    :param mut_file: path to mutations file
+    :param tmp_dir: path to temp folder
+    :param args: parsed argparse.ArgumentParser
+    :return: path to preprocessed MAF, NCBI build
+    """
+    maf = pd.read_csv(mut_file, sep="\t", comment="#", dtype=str)
+    # Get NCBI build from MAF file
+    if not args.ncbi_build:
+        ncbi_build = get_ncbi_build(maf)
+    else:
+        ncbi_build = args.ncbi_build
+    print(f"Using ncbi build: {ncbi_build}.")
+    # Determine alt allele
+    alt_al = args.alt_allele
+    if not alt_al:
+        alt_al = determine_alternative_allele(maf)
+    print(f"Using '{alt_al}' as alternative allele.")
+    # Remap maf file using important parts, as cbioportal MAF format does not conform to
+    #  GFC MAF file format (column "Consequence" is not placed on the correct location).
+    headers = ["Chromosome", "Start_Position", "End_Position", "Reference_Allele", alt_al, "Tumor_Sample_Barcode"]
+    maf = maf[headers]
+    new_headers = ["NA" for i in range(16)]
+    # SigProfilerMatrixGenerator selects alternative allele by location, so the header doesn't matter
+    for i, j in enumerate([4, 5, 6, 10, 12, 15]):  
+        new_headers[j] = headers[i]
+    maf = maf.reindex(new_headers, axis=1)
+    # File needs the MAF file extension, otherwise it is not recognized by SigProfilerMatrixGenerator
+    outfile = osp.join(tmp_dir, "mutations.maf")
+    # Fill empty cells with NA string, otherwise they get parsed incorrectly by SigProfilerMatrixGenerator
+    maf.to_csv(outfile, sep="\t", index=False, na_rep="NA")
+    return outfile, ncbi_build
+
+
+def preprocess_vcf(args: argparse.Namespace, tmp_dir):
+    """Preprocess vcf files by copying them into the tmp dir
+    
+    :param args: argparse parsed arguments
+    :param tmp_dir: temporary directory
+    """
+    vcf_list = glob.glob(osp.join(args.vcf_folder, "*.vcf"))
+    for vcf in vcf_list:
+        shutil.copy(vcf, osp.join(tmp_dir, osp.basename(vcf)))
+
+
+def prepare_input(args: argparse.Namespace) -> tuple:
+    """Wrapper function for preparing input files for matrix generation
+    
+    :param args: argparse parsed arguments 
+    :return: ncbi_build, study ID
+    """
+    # Init variables
+    maf_file, study_id, ncbi_build = [""] * 3
```

Comment:
Use `None` instead of empty string

```suggestion
    maf_file, study_id, ncbi_build = (None,) * 3
```

Reply:
also, `maf_file` is created in next block, so it's obsolete to init it here

Reply:
and `study_id` is not used

Reply:
Changed to [None]. For the variables, I can remove them, but this way the function wont crash if they don't get initialized